### PR TITLE
apis/nfd: split rule processing into a separate package

### PR DESCRIPTION
**Pull request by marquiz**
This patch tidies up the nfdv1alpha1 API package by refactoring out the implementation of (NodeFeature)Rule evaluation into a separate package.

### CODE REVIEW

File: pkg/apis/nfd/v1alpha1/nodefeaturerule/expression.go [hunk 1]
```diff
@@ -158,16 +160,16 @@ func (m *MatchExpression) Match(valid bool, value interface{}) (bool, error) {
 }
 
 // MatchKeys evaluates the MatchExpression against a set of keys.
```
Comment (AhmedGrati) [inline pkg/apis/nfd/v1alpha1/nodefeaturerule/expression.go:162]:
```suggestion
// evaluateMatchExpressionKeys evaluates the MatchExpression against a set of keys.
```

File: pkg/apis/nfd/v1alpha1/nodefeaturerule/expression_test.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (AhmedGrati) [inline pkg/apis/nfd/v1alpha1/nodefeaturerule/expression_test.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: pkg/apis/nfd/v1alpha1/nodefeaturerule/expression-api_test.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (marquiz) [reply pkg/apis/nfd/v1alpha1/nodefeaturerule/expression-api_test.go:2]: This is basically just copying (or splitting) `pkg/apis/nfd/v1alpha1/expression_test.go`, not new content, so I'd rather keep the year as is (as the "year of first publication")

File: pkg/apis/nfd/v1alpha1/nodefeaturerule/expression_test.go (hunk 2 above)
Comment (marquiz) [reply pkg/apis/nfd/v1alpha1/nodefeaturerule/expression_test.go:2]: Same here, no new content, just moving old one around so I'd keep the year as is.

File: pkg/apis/nfd/v1alpha1/nodefeaturerule/expression.go (hunk 1 above)
Comment (marquiz) [reply pkg/apis/nfd/v1alpha1/nodefeaturerule/expression.go]: Thanks for noticing this. Fixed

File: pkg/nfd-master/nfd-master.go [hunk 4]
```diff
@@ -54,6 +54,7 @@ import (
 
 	"sigs.k8s.io/node-feature-discovery/pkg/apihelper"
 	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1"
+	nodefeaturerule "sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1/nodefeaturerule"
```
Comment (ArangoGutierrez) [inline pkg/nfd-master/nfd-master.go:57]: I think the alias is not necessary since you are using the same pkg name

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: pkg/nfd-master/nfd-master.go (hunk 4 above)
Comment (marquiz) [reply pkg/nfd-master/nfd-master.go]: True, removed

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**